src/megaclient.cpp

```diff
@@ -9592,6 +9588,30 @@ error MegaClient::addsync(string* rootpath, const char* debris, string* localdeb
         }
     }
 
+    if (isinshare)
+    {
+        *isinshare = inshare;
+    }
+    return API_OK;
+#else
+    return API_EINCOMPLETE;
+#endif
+}
+
+// check sync path, add sync if folder
+// disallow nested syncs (there is only one LocalNode pointer per node), return
+// EEXIST otherwise
```

Comment:
The errorcode in the comment is outdated with this changes. It can be EACCESS, EARGS, EINCOMPLETE, EEXIST.